[#25] 상품 수정하기 기능 추가

ethan-sh-lee opened the pull request:
상품 수정하는 기능 추가
상품 수정 시 캐시에 데이터가 존재할 경우 함께 수정

isaac1102 committed:
[#12] MySQL Replication 구현

isaac1102 committed:
[#12] enum클래스명 수정

isaac1102 committed:
[#12] enum클래스명 수정

isaac1102 committed:
[#12]읽기 작업에 대하여 Transaction을 read-only 설정

isaac1102 committed:
Merge pull request #23 from f-lab-edu/feature/12

[#12] MySQL Replication 구현

ethan-sh-lee committed:
기능 : 상품 수정하기 기능 추가

ethan-sh-lee committed:
기능 : 상품 수정하기에 캐시 수정 기능 추가

ethan-sh-lee committed:
기능 : 상품 삭제하기 기능 추가

ethan-sh-lee committed:
기능 : 상품 삭제하기 기능에 캐시 추가

상품 삭제하기 요청이 들어왔을 때 해당 상품에 대한 캐시가 있을경우 해당 캐시를 삭제하는 기능 추가

f-lab-dev commented on src/main/java/com/project/smartstore/controller/ProductController.java:
```diff
@@ -38,9 +40,15 @@ public PaginationListDto getProductList(
     return productService.getProduct(productId);
   }
+
+  @CachePut(value = "product", key = "#productId")
```
캐시와 관련된건 비즈니스 로직이 들어가는 레이어에 있는게 좋지 않을까요?

Commit:
[#29]MySQL replication구현에 따른 MyBatis alias 수정

두개의 DataSource를 하나로 통합(`proxyDataSour`)하였고, `proxyDataSour`를 주입받는
`SqlSessionFactory`에서 공동의 `alias package` 경로를 지정하였습니다.

Commit:
[#29] MyBatis aliases설정 적용

isaac1102 committed:
Merge pull request #30 from f-lab-edu/feature/29

[#29]MySQL replication구현에 따른 MyBatis alias 수정

isaac1102 committed:
[#24] github-jenkins 연계테스트1

ethan-sh-lee committed:
스타일 : ProductMapper.xml 들여쓰기 수정

ethan-sh-lee committed:
Merge pull request #28 from f-lab-edu/feature/27

[#27] 상품 삭제하기 기능 추가

ethan-sh-lee committed:
빌드 : 패키징 파일 포맷을 war에서 jar로 변경

ethan-sh-lee committed:
Merge pull request #33 from f-lab-edu/feature/32

[32] 배포 파일 포맷 변경

ethan-sh-lee committed:
리팩토링 : 캐시 어노테이션을 컨트롤러단에서 서비스단으로 이동

ethan-sh-lee committed:
Merge branch 'develop' of https://github.com/f-lab-edu/smart-store into feature/25

ethan-sh-lee committed:
Merge branch 'feature/25' of https://github.com/f-lab-edu/smart-store into feature/25
